# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/TypedArrayObject.cpp
# Commit: 2d444b132a3d
# Full Hash: 2d444b132a3dba9af23fa8e477790e28a235aa0a
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2021-09-09 15:22:41
# Regressor Bug: 1720422
# File Overlap Count: 2
# Description:
#   Bug 1720422 - Convert the JS_GetObjectAs(type)Array APIs r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D119859
# ==============================================================================

diff -r 456fc09d6816 -r 2d444b132a3d js/src/vm/TypedArrayObject.cpp
--- a/js/src/vm/TypedArrayObject.cpp	Wed Sep 08 23:46:59 2021 +0000
+++ b/js/src/vm/TypedArrayObject.cpp	Wed Sep 08 23:46:59 2021 +0000
@@ -2678,20 +2678,6 @@
     return js::Unwrap##Name##Array(obj) != nullptr;                           \
   }                                                                           \
                                                                               \
-  JS_PUBLIC_API JSObject* JS_GetObjectAs##Name##Array(                        \
-      JSObject* obj, size_t* length, bool* isShared, ExternalType** data) {   \
-    obj = js::Unwrap##Name##Array(obj);                                       \
-    if (!obj) {                                                               \
-      return nullptr;                                                         \
-    }                                                                         \
-    TypedArrayObject* tarr = &obj->as<TypedArrayObject>();                    \
-    *length = tarr->length();                                                 \
-    *isShared = tarr->isSharedMemory();                                       \
-    *data = static_cast<ExternalType*>(tarr->dataPointerEither().unwrap(      \
-        /*safe - caller sees isShared flag*/));                               \
-    return obj;                                                               \
-  }                                                                           \
-                                                                              \
   JS_PUBLIC_API ExternalType* JS_Get##Name##ArrayLengthAndData(               \
       JSObject* obj, size_t* length, bool* isSharedMemory,                    \
       const JS::AutoRequireNoGC& nogc) {                                      \
@@ -2708,6 +2694,19 @@
     size_t length;                                                            \
     return JS_Get##Name##ArrayLengthAndData(obj, &length, isSharedMemory,     \
                                             nogc);                            \
+  }                                                                           \
+  JS_PUBLIC_API JSObject* JS_GetObjectAs##Name##Array(                        \
+      JSObject* obj, size_t* length, bool* isShared, ExternalType** data) {   \
+    obj = js::Unwrap##Name##Array(obj);                                       \
+    if (!obj) {                                                               \
+      return nullptr;                                                         \
+    }                                                                         \
+    TypedArrayObject* tarr = &obj->as<TypedArrayObject>();                    \
+    *length = tarr->length();                                                 \
+    *isShared = tarr->isSharedMemory();                                       \
+    *data = static_cast<ExternalType*>(tarr->dataPointerEither().unwrap(      \
+        /*safe - caller sees isShared flag*/));                               \
+    return obj;                                                               \
   }
 
 JS_FOR_EACH_TYPED_ARRAY(IMPL_TYPED_ARRAY_JSAPI_CONSTRUCTORS)
